### ISSUE TITLE
I drew a sequence diagram about signaling 101

### ISSUE THREAD

**Issue by WofWca** [webrtc-for-the-curious/webrtc-for-the-curious#327]
![pako_eNq9Uz1PwzAQ_SuHF5Y2A2OEKgFdYEARQWLJcrUviZFzDrZTqKr-dy4fA2pZwZP9fL73nv18VNobUrmK9DEQa9pabAJ2FYOMHkOy2vbICe6c1XQJl7ZhdJabksKeAmCE0ncE9GVjEhi077qBrcZkPYNukZnc6nYXNpQ1GRRE4amEOJ2e4L015EGjcxMqrNcRnkVl9h5_1vkAtCc](https://github.com/user-attachments/assets/48a5dad3-bd10-4c0b-b449-b14ab1f96065)

I think it would be nice for this section https://webrtcforthecurious.com/docs/01-what-why-and-how/#signaling-how-peers-find-each-other-in-webrtc , as a follow-up to #301.
I also thought about adding this to the "Signaling" chapter, but IMO the more suitable place is where an attempt at describing signaling first takes place.
A diagram for the signaling chapter could then be more in-depth and complex, something like in [this article](https://meetrix.io/blog/webrtc/what-is-webrtc.html).

Here it is [in the Mermaid code form](https://mermaid.live/edit#pako:eNq9Uz1PwzAQ_SuHF5Y2A2OEKgFdYEARQWLJcrUviZFzDrZTqKr-dy4fA2pZwZP9fL73nv18VNobUrmK9DEQa9pabAJ2FYOMHkOy2vbICe6c1XQJl7ZhdJabksKeAmCE0ncE9GVjEhi077qBrcZkPYNukZnc6nYXNpQ1GRRE4amEOJ2e4L015EGjcxMqrNcRnkVl9h5_1vkAtCcG6tC6q0tl9343g5Py9WZzJjWfN6R5uS3A17VAY99WXDwWazQmUIyAbKD3Ic2SWYdDPzshoapHXwS1NKXQB8tzGVDS2Ux-RioyRNf_UAvRb64FXoiR4-eITNQ2_qnryfD_kC_vPd3zG-1eXh-guCkkiJI8PTaSsKiV6ihIcoxk_zieq1RqqaNK5TI1VOPgUqUqPkkpDsmXB9YqT2GglQp-aFqV1-iirIbeiKLl4yzo6Rvwxys4).

What do you think?

**Comment by mogren** [comment]
This is a great idea, since it's somewhat unclear how this initial handshake happens. 

Sorry for the long delay on #301 btw, forgot about it for a while.

**Comment by WofWca** [comment]
> Sorry for the long delay on https://github.com/webrtc-for-the-curious/webrtc-for-the-curious/pull/301 btw, forgot about it for a while.

No worries at all!

So, shall I make an MR? Does the diagram look good, or would you adjust it?

**Comment by mogren** [comment]
The middle participant text is a bit long, what you think of something like this?

[![](https://mermaid.ink/img/pako:eNq9k0FPhEAMhf9KMxcvrAePxGyirgdNTIiYeOHSnSnQCDPYGVY3m_3vFtgYs_GqnJhX6PcetAdjgyOTm0jvI3lLG8ZGsK886IU2BYGbji0twoCS2PKAPkHJjceOfVOS7EgAI9x_ckyqgA19P3q2mDj4662sbYveU5dBx28EBZE8ltlUgB07CmCx06LCaEceqEfuflq4DdvlOHtZrddn8HwpXEQoNwWEulZp6t6qr4dihc4JxQjoHQxB0kzWtLIfJoNgSWPVk12aPdXamGQQ9ikDSvZygZ9B1Yb6-h-0gn5LrfIJjD5-TMqM5vinqefA_wM__e_5O29YyCZ4pe3zyx0UV4WOmQ6VnfpV3mSmJ9HBcTrNh-ntyqSWeqpMrreOahy7VJnKH_VRHFMo996aPMlImZEwNq3Ja-yinsbBqa_TKnyr5Fhn8WnZl3ltjl9kJCDL?type=png)](https://mermaid.live/edit#pako:eNq9k0FPhEAMhf9KMxcvrAePxGyirgdNTIiYeOHSnSnQCDPYGVY3m_3vFtgYs_GqnJhX6PcetAdjgyOTm0jvI3lLG8ZGsK886IU2BYGbji0twoCS2PKAPkHJjceOfVOS7EgAI9x_ckyqgA19P3q2mDj4662sbYveU5dBx28EBZE8ltlUgB07CmCx06LCaEceqEfuflq4DdvlOHtZrddn8HwpXEQoNwWEulZp6t6qr4dihc4JxQjoHQxB0kzWtLIfJoNgSWPVk12aPdXamGQQ9ikDSvZygZ9B1Yb6-h-0gn5LrfIJjD5-TMqM5vinqefA_wM__e_5O29YyCZ4pe3zyx0UV4WOmQ6VnfpV3mSmJ9HBcTrNh-ntyqSWeqpMrreOahy7VJnKH_VRHFMo996aPMlImZEwNq3Ja-yinsbBqa_TKnyr5Fhn8WnZl3ltjl9kJCDL)

Just updated [your diagram](https://mermaid.live/edit#pako:eNq9k0FPhEAMhf9KMxcvrAePxGyirgdNTIiYeOHSnSnQCDPYGVY3m_3vFtgYs_GqnJhX6PcetAdjgyOTm0jvI3lLG8ZGsK886IU2BYGbji0twoCS2PKAPkHJjceOfVOS7EgAI9x_ckyqgA19P3q2mDj4662sbYveU5dBx28EBZE8ltlUgB07CmCx06LCaEceqEfuflq4DdvlOHtZrddn8HwpXEQoNwWEulZp6t6qr4dihc4JxQjoHQxB0kzWtLIfJoNgSWPVk12aPdXamGQQ9ikDSvZygZ9B1Yb6-h-0gn5LrfIJjD5-TMqM5vinqefA_wM__e_5O29YyCZ4pe3zyx0UV4WOmQ6VnfpV3mSmJ9HBcTrNh-ntyqSWeqpMrreOahy7VJnKH_VRHFMo996aPMlImZEwNq3Ja-yinsbBqa_TKnyr5Fhn8WnZl3ltjl9kJCDL) a bit.

**Comment by WofWca** [comment]
I agree it might be too long. But now the "video call" part is a bit confusing.
If we go this way, I'd keep "e.g. a simple Node.js server, or even email"

**Comment by mogren** [comment]
I don't think Node.js is important, it could be written in any language. But some kind of initial communication is needed.

**Comment by WofWca** [comment]
Sure, but I just want to make it as concrete as possible.

**Comment by mogren** [comment]
I see what you mean, and I agree that mention PeerJS isn't right either. Some other suggerstions:

* Existing communication channel like chat, video call, or even email
* Existing communication channel like a message app, video call, or even email
* Existing communication channel between Alice and Bob

**Comment by WofWca** [comment]
Ok, are you fine with "e.g. a simple Node.js server, or even email"?

**Comment by mogren** [comment]
I was trying to stick to what Sean wrote in the readme and in the chapters: "This book is vendor agnostic, and we have tried to remove any conflicts of interest.". 

That's why I don't think using "Node.js" is the right expression. Why can't it be php, go, java or a server written in any other language? Isn't in better to not specify a certain language, especially when it really doesn't matter like in this case?

**Comment by WofWca** [comment]
I see. How about "e.g. a simple back-end server, or even email"?

**Comment by mogren** [comment]
Yes, I like that :)
